lib/hooks/queries/useReadyToReportMarkets.ts

```diff
@@ -0,0 +1,45 @@
+import { useQuery } from "@tanstack/react-query";
+import { MarketStatus } from "@zeitgeistpm/indexer";
+import { isFullSdk } from "@zeitgeistpm/sdk-next";
+import { isNotNull } from "@zeitgeistpm/utility/dist/null";
+import { useSdkv2 } from "../useSdkv2";
+
+export const useReadyToReportMarkets = (account?: string) => {
+  const [sdk, id] = useSdkv2();
+
+  const enabled = sdk && isFullSdk(sdk) && account;
+
+  return useQuery(
+    [id, "ready-to-report-markets", account],
+    async () => {
+      if (enabled) {
+        const closedMarketsForAccount = await sdk.indexer.markets({
+          where: {
+            oracle_eq: account,
+            status_eq: MarketStatus.Closed,
+          },
+        });
+
+        let readyToReportMarkets = (
+          await Promise.all(
+            closedMarketsForAccount.markets.map(async (market) => {
+              const stage = await sdk.model.markets.getStage(market);
```

Comment:
Could we instead just use market status? That way it will just be 2 requests rather than a request for every asset

Reply:
No status of a market will be closed also when its in graceperiod, so we have to determine the more granular stage of the market, if its in oracle reporting or open reporting. The async function `model.markets.getStage` doesnt do a lot of heavy lifting, it just fetches the current on chain time and block, but I can pass that as a param so its only fetched once up front. Other than that it shouldnt make any more requests for closed markets.

Reply:
Ah yeah, probably would be best to pull the calculation stuff out of the function so we can call it separately if we already have all of the info needed. I think it's ok for now though